maya.py

```diff
@@ -53,29 +40,45 @@ def __format__(self, *args, **kwargs):
         """Return's the datetime's format"""
         return format(self.datetime(), *args, **kwargs)
 
-    @validate_type_mayadt
+    def _validate_type_mayadt(operator):
```

Comment:
Even I was little confused whether I should move this within the function or not. The reason in favor of moving it within the class is _Error message is tightly coupled to the first argument that is `self`. Here, we verify the class of `self` with rest of the arguments. Since `self` belongs to the class, it makes sense to move it within the class. Also, as the name says `_validate_type_mayadt`, it is applicable only for MayaDT class_

But on giving a second thought on this, if we change the name of decorator to `validate_class_type_arguments`, it could be usable for other future classes as well. What are your thoughts over this?